### PR TITLE
allwinner: Add BL32 (corresponds to Trusted OS) support

### PR DESCRIPTION
**Pull request by Amit-Radur**
This pull request is an attempt to run Trusted OS (OP-TEE OS being one of them) along
side BL31 image.

This is the second version of PR  #1431, @Andre-ARM can you please have a look.

Trusted OS image can be fetched from below link:
https://github.com/Amit-Radur/Images.git

Signed-off-by: Amit Singh Tomar <amittomer25@gmail.com>

### CODE REVIEW

**Comment by davidcunado-arm** [comment]
PRs should target integration branch as per contributing guidelines.

File: docs/plat/allwinner.rst [hunk 1]
```diff
@@ -27,3 +27,23 @@ To build:
...
 .. _U-Boot documentation: http://git.denx.de/?p=u-boot.git;f=board/sunxi/README.sunxi64;hb=HEAD
+
+Trusted OS dispatcher
+=====================
+
+One can boot Trusted OS(OP-TEE OS, bl32 image) along side bl31 image on Allwinner A64.
+In order to include the 'opteed' dispatcher in the image, pass 'SPD=opteed'
+on the command line while preparing a bl31 image.
+
+Preparing the BL31 image with dispatcher
```
Comment (Andre-ARM) [inline docs/plat/allwinner.rst]: This example is not really necessary, as it just repeats what you said above. I would just write:
"..., pass `SPD=opteed` on the make command line when compiling the bl31 image and make sure the loader (SPL) loads the Trusted OS binary to the beginning of DRAM (0x40000000)."
Can we change this address? Does the Makefile support defining BL32_BASE easily?

File: plat/allwinner/common/sunxi_bl31_setup.c [hunk 2]
```diff
@@ -34,6 +35,17 @@ void bl31_early_platform_setup2(u_register_t arg0, u_register_t arg1,
...
 			       SUNXI_UART0_BAUDRATE, &console);
 
+#ifdef BL32_BASE
+	/* Populate entry point information for BL32 */
+	SET_PARAM_HEAD(&bl32_image_ep_info, PARAM_EP, VERSION_1, 0);
+	SET_SECURITY_STATE(bl32_image_ep_info.h.attr, SECURE);
+	/*
+	 * Tell BL32 where the trusted software image
+	 * is located and the entry state information
+	 */
```
Comment (Andre-ARM) [inline plat/allwinner/common/sunxi_bl31_setup.c]: This comment is a bit weird, I don't think we tell BL32 anything here, it's BL31 that reads this data. So can you please just remove it? The comment above should be sufficient.

**Comment by antonio-nino-diaz-arm** [comment]
IP Review +1

**Pull request merged**